### PR TITLE
storybook demo and small fixes

### PR DESCRIPTION
**Pull request by nabalone**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/EthnoLib/5)

### CODE REVIEW

File: components/language-picker/common/find-language/package.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
   ],
   "scripts": {
     "langtag-processing": "ts-node-esm ./language-data/langtagProcessing.ts",
-    "build": "tsup ./index.ts --format esm --dts --external react",
-    "dev": "tsup ./index.ts --format esm --watch --dts --external react",
```
Comment (nabalone) [inline components/language-picker/common/find-language/package.json:15]: I missed this earlier. `find-language` doesn't actually care about react

File: package.json [hunk 2]
```diff
@@ -12,7 +12,25 @@
...
+    "@storybook/core-server": "7.6.20",
+    "@storybook/jest": "^0.2.3",
+    "@storybook/react-vite": "7.6.20",
+    "@storybook/test-runner": "^0.13.0",
+    "@storybook/testing-library": "^0.2.2",
+    "@swc-node/register": "~1.9.1",
+    "@swc/core": "~1.5.7",
+    "@swc/helpers": "~0.5.11",
     "nx": "^19.1.2",
+    "prettier": "^2.6.2",
```
Comment (nabalone) [inline package.json:30]: Not sure how I ended up adding prettier but I think we do want it here right? bloom-player has it in devDependencies also

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 12 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/EthnoLib/5)* status: 12 of 14 files reviewed, 1 unresolved discussion (waiting on @nabalone)

___
*[`package.json` line 30 at r1](https://reviewable.io/reviews/sillsdev/EthnoLib/5#gh-1731912445:-O5Ft1oV6bAaQJeVLNzS:birgvl5) ([raw file](https://github.com/sillsdev/EthnoLib/blob/8669cd15c8e7b7353ef7a461e30884c28b498ad3/package.json#L30)):*
<details><summary><i>Previously, nabalone (Noel) wrote…</i></summary><blockquote>

Not sure how I ended up adding prettier but I think we do want it here right? bloom-player has it in devDependencies also
</blockquote></details>

Sounds right.
___
*[`tsconfig.base.json` line 1 at r1](https://reviewable.io/reviews/sillsdev/EthnoLib/5#-O5FtWs504GHeIK7lMKF:-O5FtWs504GHeIK7lMKG:b-yj44us) ([raw file](https://github.com/sillsdev/EthnoLib/blob/8669cd15c8e7b7353ef7a461e30884c28b498ad3/tsconfig.base.json#L1)):*
> ```JSON
> {
> ```

Is this file being used?
I would have expected with a file like this that both tsconfig.json an tsconfig.storybook.json would extend it or something.

**Review by nabalone — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/EthnoLib/5)* status: 12 of 14 files reviewed, 1 unresolved discussion (waiting on @andrew-polk)

___
*[`tsconfig.base.json` line 1 at r1](https://reviewable.io/reviews/sillsdev/EthnoLib/5#-O5FtWs504GHeIK7lMKF:-O5Jvahp30RSmt1XHgI9:b-1f5c55) ([raw file](https://github.com/sillsdev/EthnoLib/blob/8669cd15c8e7b7353ef7a461e30884c28b498ad3/tsconfig.base.json#L1)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Is this file being used?
I would have expected with a file like this that both tsconfig.json an tsconfig.storybook.json would extend it or something.
</blockquote></details>

If I delete the file I get the following error: 
```
The "@nx/storybook/plugin" plugin threw an error while creating nodes from components/language-picker/react/language-chooser-react-mui/.storybook/main.ts:
    C:\Users\Noel\Documents\work\EthnoLib\components\language-picker\react\language-chooser-react-mui\.storybook\main.ts:1
    import type { StorybookConfig } from "@storybook/react-vite";
```
Maybe NX is using it?

**Review by JohnThomson — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/EthnoLib/5)* status: 12 of 14 files reviewed, 1 unresolved discussion (waiting on @andrew-polk)

___
*[`tsconfig.base.json` line 1 at r1](https://reviewable.io/reviews/sillsdev/EthnoLib/5#-O5FtWs504GHeIK7lMKF:-O5KswTP6sVCk7cMmvo-:b-kk7lr3) ([raw file](https://github.com/sillsdev/EthnoLib/blob/8669cd15c8e7b7353ef7a461e30884c28b498ad3/tsconfig.base.json#L1)):*
<details><summary><i>Previously, nabalone (Noel) wrote…</i></summary><blockquote>

If I delete the file I get the following error: 
```
The "@nx/storybook/plugin" plugin threw an error while creating nodes from components/language-picker/react/language-chooser-react-mui/.storybook/main.ts:
    C:\Users\Noel\Documents\work\EthnoLib\components\language-picker\react\language-chooser-react-mui\.storybook\main.ts:1
    import type { StorybookConfig } from "@storybook/react-vite";
```
Maybe NX is using it?
</blockquote></details>

That seems like sufficient reason to keep it, though it would be nice to understand better what is affected by changes to this file.
None of the changes mention it, so it is strange that it would be needed now and not before, and strange that it got added at all without the author really knowing why. Maybe something new is pulling in a file which hasn't been changed but which uses it? Do you find anything by doing a global search of your project for the file name?
In any case, since Andrew was happy with everything else here, I don't see any reason not to merge it. But I'll wait to do so in case you want to do any more research.

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 2 of 14 files at r1.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/EthnoLib/5)* status: all files reviewed, 1 unresolved discussion (waiting on @no)

___
*[`tsconfig.base.json` line 1 at r1](https://reviewable.io/reviews/sillsdev/EthnoLib/5#-O5FtWs504GHeIK7lMKF:-O76Ig8o4pirwk0NMIZ9:b-ea0snb) ([raw file](https://github.com/sillsdev/EthnoLib/blob/8669cd15c8e7b7353ef7a461e30884c28b498ad3/tsconfig.base.json#L1)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

That seems like sufficient reason to keep it, though it would be nice to understand better what is affected by changes to this file.
None of the changes mention it, so it is strange that it would be needed now and not before, and strange that it got added at all without the author really knowing why. Maybe something new is pulling in a file which hasn't been changed but which uses it? Do you find anything by doing a global search of your project for the file name?
In any case, since Andrew was happy with everything else here, I don't see any reason not to merge it. But I'll wait to do so in case you want to do any more research.
</blockquote></details>

Was any progress made on this? Or do we care?
Happy to merge if you are satisfied, @nabalone.

**Review by nabalone — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/EthnoLib/5)* status: all files reviewed, 1 unresolved discussion (waiting on @andrew-polk and @JohnThomson)

___
*[`tsconfig.base.json` line 1 at r1](https://reviewable.io/reviews/sillsdev/EthnoLib/5#-O5FtWs504GHeIK7lMKF:-O7G5bSS43Xup3474MMc:b44g4pv) ([raw file](https://github.com/sillsdev/EthnoLib/blob/8669cd15c8e7b7353ef7a461e30884c28b498ad3/tsconfig.base.json#L1)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Was any progress made on this? Or do we care?
Happy to merge if you are satisfied, @nabalone.
</blockquote></details>

I think we can merge. I ended up adding this same file with the same parameters on a different branch anyway, as it ends up being necessary for our publishing config. So we will have merge conflicts later but easy to resolve

**Pull request merged**